cloudapp/src/app/excel-export/excel-export-service.ts

```diff
@@ -0,0 +1,137 @@
+import { EventEmitter, Injectable } from '@angular/core'
+
+import { ColumnOption } from '../column-options'
+import { ColumnDefinition, COLUMNS_DEFINITIONS } from '../column-definitions'
+
+import * as XLSX from 'sheetjs-style'
+import * as FileSaver from 'file-saver'
+import { RequestedResource, RequestedResourcesService } from '../requested-resources-service/requested-resources-service'
+
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExcelExportService {
+  private static readonly FILE_NAME: string = 'requested-resources'
+  private static readonly EXCEL_TYPE: string = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;charset=UTF-8'
+  private static readonly EXCEL_EXTENSION: string = '.xlsx'
+
+  private static readonly PAGE_SIZE: number = 100
+
+  constructor(
+    private requestedResourcesService: RequestedResourcesService
+  ) { }
+
+
+  async generateExcel(circDeskCode: string, libraryCode: string, columnOptions: ColumnOption[]) {
+
+    const columnDefinitions: ColumnDefinition[] = this.getColumnDefinitions(columnOptions)
+
+    this.requestedResourcesService.findRequestedResources(
+		circDeskCode, 
+		libraryCode, 
+		ExcelExportService.PAGE_SIZE, 
+		null, 
+		(count:number) => { console.log('completed', count) }, 
+		(err:Error) => { console.log('error', err) })
+      .then(resources => {
+        const data: string[][] = this.createOutputFormat(resources, columnDefinitions)
+        const worksheet: XLSX.WorkSheet = XLSX.utils.aoa_to_sheet(data)
+        this.setCellWith(worksheet, columnDefinitions)
+        this.setCellStyles(worksheet)
+        const workbook: XLSX.WorkBook = { Sheets: { 'data': worksheet }, SheetNames: ['data'] }
+        const excelBuffer: any = XLSX.write(workbook, { bookType: 'xlsx', type: 'array' })
+        this.saveAsExcelFile(excelBuffer)
+      })
+      .catch(e => console.log(e))
```

Comment:
Please don't swollow exception. Let it rise out of here, catch in the main component and show an alert.

Reply:
Fixed in commit 986ce86. No exception catching there anymore, but try catch block with alert in main component, as you suggested. Thank you.